gpu/src/cuda/field/element.rs

```diff
@@ -0,0 +1,42 @@
+use cudarc::driver::safe::DeviceRepr;
+use lambdaworks_math::field::{element::FieldElement, traits::IsField};
+
+use core::ffi;
+
+#[derive(Clone)]
+pub struct CUDAFieldElement<F: IsField> {
+    value: F::BaseType,
+}
+
+impl<F: IsField> CUDAFieldElement<F> {
+    /// Returns the underlying `value`
+    pub fn value(&self) -> &F::BaseType {
+        &self.value
+    }
+}
+
+impl<F: IsField> Default for CUDAFieldElement<F> {
+    fn default() -> Self {
+        Self { value: F::zero() }
+    }
+}
+
+unsafe impl<F: IsField> DeviceRepr for CUDAFieldElement<F> {
```

Comment:
couldn't you implement this trait for FFTField instead and not create this abstraction?